### DISCUSSION TITLE
Why is SmoothGraphics a plugin?

### DISCUSSION THREAD

**Discussion by humbleCoder1990** [pixijs/pixijs#7732] [Q&A]
![image](https://user-images.githubusercontent.com/87828192/130335157-12198c66-704e-4d48-9812-3e85d224fb4a.png)

I'm really enjoying SmoothGraphics.
I hope SmoothGraphics should be built-in, or replace Graphics.
Is there any downside of SmoothGraphics compared Graphics?
Do SmoothGraphics use more CPU or GPU ?

**Comment by bigtimebuddy** [comment]
Thanks for the feedback. @ivanpopelyshev did a great job with smooth graphics. The reason it's a plugin is that it's the best way to test out ideas and this is a very new plugin and a new approach to antialiasing. There are still some incomplete features (no smoothing for fills), but I could see a future where we add it as the default or as an option.

**Reply by humbleCoder1990** [reply]
I really got it "The reason it's a plugin is that it's the best way to test out ideas"

Thanks Matt.

Because I see no performance problem with Smooth Graphics,too
It works really well.
There will be users like me who try to draw most of things in PIXI instead of using image files like PNG (kind of Vector style).
I switched to Pixi from React to make GUI Music app, (not game)
I'm very much happy with PIXI, I'll never go back to React.
PIXI is very best engine to create any GUI app.
If the "antialiasing' algorithm will be solved, The last puzzle will be solved.
Graphics's fill also doesn't look good to draw rounded or circle.
SmoothGraphics is really really cool, I hope  this plugin development shouldn't be stopped.

Thanks always It's amazing!

**Comment by ivanpopelyshev** [comment]
The plugin was supposed to be commercial project for huge amount of $$$, however when I saw results, I refused. 10 years of webgl and that's the first time we see canvas2d-grade of quality and performance for lines.

I know how to make both implementations (vanilla & smooth) to work in same class. 

I made architecture that has empty places for many possible hacks, and its filled only by 10% of what i want. Optimizations, featurse, there a lot of things to do. For that, I need to be able re-publish this thing whenever i want - it wont be possible when i put it to main pixi repo :( Well, I know how to make it pluggable, so people in main version will have some of features, and latest will be available in my repo, but that's extra time i dont have right now
